game/gamestate.go

```diff
@@ -147,3 +308,149 @@ func (gs *GameState) AreAdjacent(cantonID1, cantonID2 int) bool {
 	}
 	return false
 }
+
+func (gs *GameState) Delta() map[string]any {
+	if gs.LastMove == nil {
+		return nil
+	}
+	gameMove := gs.LastMove.(*GameMove)
+	return map[string]any{
+		"ActionType":   gameMove.ActionType,
+		"FromCantonID": gameMove.FromCantonID,
+		"ToCantonID":   gameMove.ToCantonID,
+		"NumTroops":    gameMove.NumTroops,
+	}
+}
+
+func (gs *GameState) Play(move Move) State {
+	newGs := gs.Copy()
+	gameMove := move.(*GameMove)
+
+	switch gs.Phase {
+	case ReinforcementPhase:
+		if gameMove.ActionType == ReinforceAction {
+			// Apply reinforcement move
+			newGs.TroopCounts[gameMove.ToCantonID] += gameMove.NumTroops
+			// Subtract placed troops from troops to place
+			newGs.TroopsToPlace -= gameMove.NumTroops
+			if newGs.TroopsToPlace <= 0 {
+				newGs.AdvancePhase()
+			}
+		} else {
+			// Invalid action for this phase
+			return newGs

```

Comment:
please throw error here and in other places like this (silencing errors will make integrating and debugging our work difficult)